hudi-client/src/main/java/org/apache/hudi/index/bloom/BucketizedBloomCheckPartitioner.java

```diff
@@ -145,7 +145,7 @@ public int getPartition(Object key) {
     final Pair<String, String> parts = (Pair<String, String>) key;
     final long hashOfKey = NumericUtils.getMessageDigestHash("MD5", parts.getRight());
     final List<Integer> candidatePartitions = fileGroupToPartitions.get(parts.getLeft());
-    final int idx = (int) Math.floorMod(hashOfKey, candidatePartitions.size());
+    final int idx = (int) Math.floorMod((int) hashOfKey, candidatePartitions.size());
```

Comment:
is the cast to `(int)` necessary still before assigning to `idx`

Reply:
Hey @vinothchandar  If you take a look at the original discussion https://github.com/apache/incubator-hudi/issues/1455, there are different interface in java8 and other versions of java. The consolidate that difference, cast both to int or long will make less room for runtime error.